mysql-test/main/master_retry_count_basic.test

```diff
@@ -0,0 +1,80 @@
+# MDEV-25674: Test the `Master_Retry_Count` field of
+# CHANGE MASTER [name] TO and SHOW SLAVE [name] STATUS & co. (no feature testing)
+# Two connections tests that the field is now per-connection.
+--source include/have_perfschema.inc
+
+--echo # Use `--master-retry-count` when not specified
+CHANGE MASTER 'named' TO master_host='example.com';
+CHANGE MASTER TO master_host='127.0.0.1', master_ssl_verify_server_cert=0;
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # Replace when specified
+CHANGE MASTER 'named' TO master_retry_count=11;
+# Default master does not replace named master
+CHANGE MASTER TO master_retry_count=10;
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # Conventional views also show the configuations
+--let $all_slaves_status= 1
+--let $status_items= Connection_name, Master_Retry_Count
+--source include/show_slave_status.inc
+--let $all_slaves_status= 0
+--let $status_items= Master_Retry_Count
+--source include/show_slave_status.inc
+--let $slave_name= 'named'
+--source include/show_slave_status.inc
+SELECT CHANNEL_NAME, CONNECTION_RETRY_COUNT
+  FROM performance_schema.replication_connection_configuration;
+
+--echo # Restore specified config on restart
+--let $restart_parameters= --skip-slave-start
+--source include/restart_mysqld.inc # not_embedded
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # Keep specified config on RESET REPLICA
+RESET REPLICA 'named';
+RESET REPLICA;
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # Don't replace when not specified
+CHANGE MASTER TO master_user='root';
+CHANGE MASTER 'named' TO master_user='root';
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # 0 internally means "not specified"
+CHANGE MASTER TO master_retry_count=0;
+CHANGE MASTER 'named' TO master_retry_count=0;
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # Truncates decimals
+CHANGE MASTER TO master_retry_count=0.5;
+CHANGE MASTER 'named' TO master_retry_count=0.5;
+SELECT Connection_name, Master_Retry_Count
+  FROM information_schema.SLAVE_STATUS;
+
+--echo # Caps values (such as UINT64_MAX + 1) to `--master-retry-count`'s max
+CHANGE MASTER TO master_retry_count=18446744073709551616;
+CHANGE MASTER 'named' TO master_retry_count=18446744073709551616;
+SELECT Connection_name
+  FROM information_schema.SLAVE_STATUS
+  WHERE Master_Retry_Count IN (2147483647, 18446744073709551615);
+
```

Comment:
This fails on [amd64-windows](https://buildbot.mariadb.org/#/builders/234/builds/36722)… It could very well be that we never tested edge-case behaviors[^MDEV-35948] at all 😬.
```suggestion
--echo # Caps values (such as UINT64_MAX + 1) to `--master-retry-count`'s max
CHANGE MASTER TO master_retry_count=18446744073709551616;
CHANGE MASTER 'named' TO master_retry_count=18446744073709551616;
SELECT Connection_name
  FROM information_schema.SLAVE_STATUS
  WHERE Master_Retry_Count IN (2147483647, 18446744073709551615);

```

[^MDEV-35948]: such as [MDEV-35948 for SHOW REPLICA STATUS](https://jira.mariadb.org/browse/MDEV-35948)

Reply:
```suggestion
--echo # Caps values (such as UINT64_MAX + 1) to `--master-retry-count`'s max
CHANGE MASTER TO master_retry_count=18446744073709551616;
CHANGE MASTER 'named' TO master_retry_count=18446744073709551616;
SELECT Connection_name
  FROM information_schema.SLAVE_STATUS
  WHERE Master_Retry_Count IN (4294967295, 18446744073709551615);

```

Reply:
If this doesn’t fix the Windows difference then I’m throwing this UB case outta here.

Reply:
Oh interesting.. do you have a Windows box to debug and see what is going on? If not, mine is packed up atm, but I can check it out in a couple weeks. Maybe in the mean-time, just comment it out as a TODO for MDEV-35948?

Reply:
Well I do but not really. My computer is Windows but I primarily almost always code in its WSL environment. On native Windows’s side I downloaded an MinGW but I’m not building MariaDB with it.